src/extension/commandPaletteHandler.ts

```diff
@@ -111,16 +125,27 @@ export class CommandPaletteHandler {
                         appLauncher.setReactNativeVersions(versions);
                         return this.executeCommandInContext("runAndroid", appLauncher.getWorkspaceFolder(), () => {
                             const platform = <AndroidPlatform>this.createPlatform(appLauncher, "android", AndroidPlatform, target);
-                            return platform.beforeStartPackager()
-                                .then(() => {
-                                    return platform.startPackager();
-                                })
-                                .then(() => {
-                                    return platform.runApp(/*shouldLaunchInAllDevices*/true);
-                                })
-                                .then(() => {
-                                    return platform.disableJSDebuggingMode();
-                                });
+                            return new Promise((resolve) => {
+                                if (target === "simulator") {
+                                    resolve(platform.startEmulator(target));
+                                }
+                                else resolve(null);
+                            })
+                            .then((emulator: IAndroidEmulator | null) => {
+                                if (emulator) {
+                                    GeneralMobilePlatform.setRunArgument(platform.getRunArguments(), "--deviceId", emulator.id);
+                                }
+                            })
```

Comment:
May we replace that potion of code by async function where we firstly await for emulator execution and then change runArguments? That representation will look more compact and will allow us to hide redundant Promise.